src/Services/WorkflowService.php

```diff
@@ -180,20 +180,19 @@ public function getWorkflowFor($item, $includeComplete = false)
 
         if ($item instanceof WorkflowAction) {
             $id = $item->WorkflowID;
-            return DataObject::get_by_id(WorkflowInstance::class, $id);
+            return WorkflowInstance::get()->setUseCache(true)->byID($id);
         } elseif (is_object($item) && ($item->hasExtension(WorkflowApplicable::class)
                 || $item->hasExtension(FileWorkflowApplicable::class))
         ) {
-            $filter = sprintf(
-                '"TargetClass" = \'%s\' AND "TargetID" = %d',
-                Convert::raw2sql(DataObject::getSchema()->baseDataClass($item)),
-                $item->ID
-            );
-            $complete = $includeComplete ? 'OR "WorkflowStatus" = \'Complete\' ' : '';
-            return DataObject::get_one(
-                WorkflowInstance::class,
-                $filter . ' AND ("WorkflowStatus" = \'Active\' OR "WorkflowStatus"=\'Paused\' ' . $complete . ')'
-            );
+            $statuses = ['Active', 'Paused'];
+            if ($includeComplete) {
+                $statuses[] = 'Complete';
+            }
+            return WorkflowInstance::get()->setUseCache(true)->filter([
+                'TargetClass' => DataObject::getSchema()->baseDataClass($item),
+                'TargetID' => $item->ID,
+                'WorkflowStatus' => $statuses,
+            ])->first();
```

Comment:
`get_one()` uses raw SQL in a `where` - but there's no need for that. We can just use a safe `filter` instead.